### PR TITLE
Better logging for Kube pod failures

### PR DESCRIPTION
**Pull request by thedanvail**
- Handles the error more gracefully if we receive from Kube's API when trying to spin up a pod
- Updated CHANGELOG with new changes

### CODE REVIEW

**Comment by thedanvail** [comment]
Hopefully this is the functionality that was requested from #110 - if not, can still make changes. Thought about using `Result` and adding onto `error.rs`, but neither seemed to quite fit the bill for this in my opinion.

**Review by aviramha — CHANGES_REQUESTED** [review]
Thanks for your contribution! See comments inline.

**Comment by thedanvail** [comment]
All right, I've switched over to using `anyhow` for attaching context to these errors. I'm signing off for the night, but should any additional changes be needed, please do not hesitate to let me know and I'll get on them tomorrow evening or possibly Wednesday.

**Comment by thedanvail** [comment]
I see some checks failed, I'll look at those in the morning

**Comment by aviramha** [comment]
LGTM, @meowjesty can you review this please?

**Review by aviramha — APPROVED** [review]

File: mirrord-layer/src/pod_api.rs
Comment (meowjesty) [inline mirrord-layer/src/pod_api.rs:61]: I would suggest returning just `PortForwarder` here, and change `unwrap`s inside this function to `expect` with a nice message giving additional context.

If this function fails we're not handling a potential recovery yet, so it's better to just crash on failures.

File: mirrord-layer/src/pod_api.rs
Comment (meowjesty) [inline mirrord-layer/src/pod_api.rs:64]: For example here, I would suggest changing it to:

```rust
... = Config.infer().await.expect("Failed to infer config!");
```

Or some nicer message (I'm lacking in creativity here :P).

File: mirrord-layer/src/pod_api.rs
Comment (meowjesty) [inline mirrord-layer/src/pod_api.rs:166]: And here you change this `with_context` to `expect`(same for every similar piece of code).

**Review by meowjesty — CHANGES_REQUESTED** [review]
Hey Dan, thanks for the help in improving this! The lack of context kinda tripped me over a few times.

My main suggestion comes in the form that we probably don't need to have a `Result` here, as we just crash when this fails anyway. Just change some of the `unwrap`s into `expect`s (also no need to change every single one).

File: mirrord-layer/src/pod_api.rs
Comment (aviramha) [reply mirrord-layer/src/pod_api.rs:166]: I don't think we should call expect from this part of the code.. rather return an error and the top function should decide whether to panic (in this case, it should because it's a ctor function and can't do anything else but panic)

Comment (meowjesty) [reply mirrord-layer/src/pod_api.rs:166]: Hey Dan, talked with aviram, and we decided to go for the `Result` approach (sorry about potential confusion).

Comment (thedanvail) [reply mirrord-layer/src/pod_api.rs:166]: Haha it's all good, friends. Just to make sure I'm up to date - are there any changes you need before we merge this and I get started on #62 ? I updated the branch to be in line with `upstream/main`, so y'all just let me know if anything needs changing :)

File: mirrord-layer/src/pod_api.rs
Comment (aviramha) [inline mirrord-layer/src/pod_api.rs:171]: you lack `?` at the end of the line to make it return the error.

**Review by aviramha — CHANGES_REQUESTED** [review]
Last change and we can merge it :)

File: CHANGELOG.md
Comment (thedanvail) [reply CHANGELOG.md]: It is done, my captain!

File: mirrord-layer/src/pod_api.rs
Comment (thedanvail) [reply mirrord-layer/src/pod_api.rs]: Your wish is my command

**Comment by aviramha** [comment]
Great, can you merge main into your branch so tests will run before merging?
Thanks!

**Review by eyalb181 — APPROVED** [review]

**Comment by eyalb181** [comment]
Tests are passing, the e2e are just a bit flaky at the moment (need to look into it). Merging to main, thanks again @thedanvail !

**Pull request merged**
